src/Databases/DatabaseOnDisk.cpp

```diff
@@ -689,15 +689,58 @@ ASTPtr DatabaseOnDisk::parseQueryFromMetadata(
 ASTPtr DatabaseOnDisk::getCreateQueryFromMetadata(const String & database_metadata_path, bool throw_on_error) const
 {
     ASTPtr ast = parseQueryFromMetadata(log, getContext(), database_metadata_path, throw_on_error);
-
     if (ast)
     {
         auto & ast_create_query = ast->as<ASTCreateQuery &>();
         ast_create_query.attach = false;
         ast_create_query.database = getDatabaseName();
     }
-
     return ast;
 }
 
+/// Daisy: starts.
+const DatabaseOnDisk::MetaDataCacheElem* DatabaseOnDisk::findCache(const String & key) const
+{
+    std::shared_lock lock(cache_rw_mutex);
+    auto iter = metadata_cache.find(key);
+    if (iter != metadata_cache.end())
+        return &(iter->second);
+    return nullptr;
+}
+
+void DatabaseOnDisk::setCache(const String & key, const DatabaseOnDisk::MetaDataCacheElem& value) const
+{
+    std::lock_guard lock(cache_rw_mutex);
+    metadata_cache[key] = value;
+}
+
+
+#if defined(_WIN32)
```

Comment:
Can you use C++ standard library https://en.cppreference.com/w/cpp/filesystem/last_write_time ?